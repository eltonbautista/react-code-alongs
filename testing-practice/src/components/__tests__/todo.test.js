import { render, screen, cleanUp, queryByTestId, getByTestId, cleanup,  } from '@testing-library/react';
import React from "react";
import { Renderer } from 'react-dom';
import renderer from 'react-test-renderer';
// import { ... } from "@testing-library/react";
import "@testing-library/jest-dom";  // optional
import userEvent from "@testing-library/user-event";
import Todo from '../todo';

afterEach(() => {
  cleanup();
})

test('should render non-completed todo component', () => {
  const todo = {id: 1, title: 'wash dishes', completed: false,};
  render(<Todo todo={todo}/>);
  const todoElement = screen.getByTestId('todo-1');
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent('wash dishes');
})

test('should render completed todo component', () => {
  const todo = {id: 2, title: 'wash car', completed: true,};
  render(<Todo todo={todo}/>);
  const todoElement = screen.getByTestId('todo-2');
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent('wash car');
  expect(todoElement).toContainHTML('<strike><h1>wash car</h1></strike>');
})

test('matches snapshot', () => {
  const todo = {id: 1, title: 'wash dishes', completed: false,};
  const tree = renderer.create(<Todo todo={todo}/>).toJSON();
  expect(tree).toMatchSnapshot();
})