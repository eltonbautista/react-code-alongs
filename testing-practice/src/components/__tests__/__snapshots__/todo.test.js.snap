// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    wash dishes
  </h1>
</div>
`;
